packages/aws-cdk/lib/api/logs/find-cloudwatch-logs.ts

```diff
@@ -0,0 +1,101 @@
+import * as cxapi from '@aws-cdk/cx-api';
+import { Mode, SdkProvider, ISDK } from '../aws-auth';
+import { prepareSdkWithLookupRoleFor } from '../cloudformation-deployments';
+import { EvaluateCloudFormationTemplate, LazyListStackResources } from '../evaluate-cloudformation-template';
+
+// resource types that have associated CloudWatch Log Groups that should _not_ be monitored
+const IGNORE_LOGS_RESOURCE_TYPES = ['AWS::EC2::FlowLog', 'AWS::CloudTrail::Trail', 'AWS::CodeBuild::Project'];
+
+// Resource types that will create a CloudWatch log group with a specific name if one is not provided.
+// The keys are CFN resource types, and the values are the names of the physical name property of that resource.
+const RESOURCE_TYPES_WITH_IMPLICIT_LOGS: { [cfnResourceType: string]: string } = {
+  'AWS::Lambda::Function': 'FunctionName',
+};
+
+export async function findCloudWatchLogGroups(
+  sdkProvider: SdkProvider,
+  stackArtifact: cxapi.CloudFormationStackArtifact,
+): Promise<{ env: cxapi.Environment, sdk: ISDK, logGroupNames: string[] }> {
```

Comment:
Declare the return struct separately please.

Reply:
Done.